python/ray/util/spark/cluster_init.py

```diff
@@ -481,9 +487,25 @@ def _setup_ray_cluster(
 
     if ray_temp_root_dir is None:
         ray_temp_root_dir = start_hook.get_default_temp_dir()
-    ray_temp_dir = os.path.join(
-        ray_temp_root_dir, f"ray-{ray_head_port}-{cluster_unique_id}"
-    )
+    if global_mode_enabled():
+        # It reads from environment variable RAY_ADDRESS, then check
+        # ray_temp_root_dir/ray/ray_current_cluster to see if the address
+        # is available, if yes then a global ray cluster is running, otherwise
+        # it will start a new glbal ray cluster.
+        ray_addr = canonicalize_bootstrap_address_or_die("auto", ray_temp_root_dir)
```

Comment:
For global mode, we should use default temp dir returned by `ray._private.utils.get_user_temp_dir`, the `ray_temp_root_dir` is used by per-notebook ray cluster.